packages/astro/src/core/endpoint/index.ts

```diff
@@ -150,9 +168,50 @@ export async function callEndpoint<MiddlewareResult = Response | EndpointOutput>
 		}
 	}
 
-	return {
-		...response,
-		type: 'simple',
-		cookies: context.cookies,
-	};
+	let body: BodyInit;
```

Comment:
All the pipeline simple object handling converges into this function starting from this line. This should also make the dev/prod  behaviour more consistent.